src/functions-and-arrays.js

```diff
@@ -1,73 +1,175 @@
 
+function sumNumbers(numbers) {
+    if (numbers.length === 0) {
+        return 0
+    } else {
+        let total = 0;
+        for (i = 0; i < numbers.length; ++i) {
+            total += numbers[i];
+        }
+        return total;
+    }
+}
+
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
+
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numbersAvg) {
+    if (numbersAvg.length === 0) {
+        return null
+    } else {
+        let total = 0;
+        for (let i = 0; i < numbersAvg.length; i++) {
+            total += numbersAvg[i];
+        }
+        let avg = total / numbersAvg.length;
+        return avg
+    }
+}
+
+
+
+
+
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
+
+function averageWordLength(wordsArr) {
+    if (wordsArr.length === 0)
+        return null;
+    else {
+        let jvalues = wordsArr.join(' ');
+        let wordCount = jvalues.split(" ").length;
+        let wordAvg = 0;
+        let avgLen = 0;
+        for (let i = 0; i < wordsArr.length; i++) {
+            wordAvg += wordsArr[i].length;
+            avgLen = wordAvg / wordCount;
+        }
+
+        console.log(avgLen);
+        return avgLen;
+    }
+}
+
+
+
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+    'crab',
+    'poison',
+    'contagious',
+    'simple',
+    'bring',
+    'sharp',
+    'playground',
+    'poison',
+    'communion',
+    'simple',
+    'bring'
 ];
 
+function uniquifyArray(data) {
+    if (wordsUnique.length == 0) {
+        return null
+    } else {
+        return data.filter((value, index) => data.indexOf(value) === index);
+    }
+}
+console.log(uniquifyArray(wordsUnique));
+
+
+
+
 // Iteration #6: Find elements
 const wordsFind = ['machine', 'subset', 'trouble', 'starting', 'matter', 'eating', 'truth', 'disobedience'];
 
+function doesWordExist() {
+    let includesTwenty = wordsFind.includes("machine");
+    return includesTwenty
+    console.log(includesTwenty)
+}
+
+doesWordExist()
+
 // Iteration #7: Count repetition
 const wordsCount = [
-  'machine',
-  'matter',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'matter',
-  'truth',
-  'disobedience',
-  'matter'
+    'machine',
+    'matter',
+    'subset',
+    'trouble',
+    'starting',
+    'matter',
+    'eating',
+    'matter',
+    'truth',
+    'disobedience',
+    'matter'
 ];
 
+function howManyTimes() {
+    if (wordsCount.length === 0 || wordsCount == [])
+        return "0"
+    else
+        for (let i = 0; i < wordsCount.length; i++) {
+            let result = 0;
+            if (wordsCount[i] === "matter") {
+                result++;
+            }
+            return result;
+        }
+}
+
```

Comment:
Same thing with function howManyTimes

`function howManyTimes(arr, word) {
  if (arr.length === 0 || arr == [])
        return "0"
    else
        for (let i = 0; i < arr.length; i++) {
            let result = 0;
            if (arr[i] === word) {
                result++;
            }
            return result;
        }
}`

`howManyTimes(wordsCount, "matter")`